### ISSUE TITLE
Utility to distribute data into subgraphs

### ISSUE THREAD

**Issue by neuneck** [PaulSchweizer/flowpipe#142]
**Is your feature request related to a problem? Please describe.**
I often have subgraphs with several nodes that will take the same input from outside the subgraph (e.g. several nodes taking the same file from a file-reading node). This is not well supported as of now, all options I came up with have drawbacks. 

**Describe the solution you'd like**
I think a way to group inputs would be useful. Such a feature could be used to clearly set up several inputs to take the same information. Connecting all these inputs to the same output can then be done transparently in a single connection operation.

I think that such a grouping might be implemented as a `GroupingPlug`, which inherits from `InputPlug`, providing methods to
* be connected up to an `OutputPlug`, thereby connecting all grouped Inputs with that output
* Add a new `InputPlug` to the group, connecting it with the `OutputPlug` if a connection already exists.
* Disconnect form the output, thereby disconnecting all inputs in the group

**Describe alternatives you've considered**
Right now, there are three options to connecting that up: 

1. Connect the outside nodes directly to the inputs of the nodes within the subgraph. This is not great, since I define the subgraphs to abstract away the internal structure as much as possible.
2. Promote all the input plugs to the subgraph. This is awkward, because the subgraph then has several inputs for the same data.
3. Introduce an extra node that trivially pipes through the data. This way, we can have a single input to the subgraph, however, the extra node makes the graph unnecessarily complicated and, when caching outputs, increases the memory footprint.

**Additional context**
I think such a feature would be useful in all graphs that fan out widely. Right after instantiating a node, one could add the inputs this node shares with other nodes to the group, keeping that behavior neatly in one place. Still, the connection code would be much less boilerplate, since I could just connect `my_file_reader_node.outputs["file"] >> file_consumers`.
By being a plug itself, my suggested implementation above could provide the functionality I'm after: Group all inputs for the same outside data inside a subgraph and expose the bundle as a single unit.

Some illustrations:
Options 1 or 2 of the alternatives yield such a behavior:
![flowpipe_1](https://user-images.githubusercontent.com/23141072/97970271-8e67e800-1dc1-11eb-901d-e250d14e04e9.png)

Option 3 adds a node to the graph, that doesn't do very much and caches the data redundantly, if the output cashing is not disabled:
![flowpipe_2](https://user-images.githubusercontent.com/23141072/97970276-9162d880-1dc1-11eb-83ad-0ef46c15e7e9.png)

The requested feature would provide some utility for the circle in this diagram:
![flowpipe_3](https://user-images.githubusercontent.com/23141072/97970280-932c9c00-1dc1-11eb-8d9d-f8ff6a65475d.png)

**Comment by PaulSchweizer** [comment]
@neuneck , sounds perfectly reasonable, thanks for the detailed explanation. 
So basically we would add plugs to the graph (in- and outputs) and wire them up with the nodes inside. I think we talked about an option like this when we implemented the subgraphs but decided against it because it wasn't crucial. Now that we have subgraphs working and people are using them, I think we're in a good position to enhance them and implement this.
Let me know if you want to implement it, otherwise I could lend a hand and take look at it.
